pkg/dashboard/functiontemplates/generator/generator.go

```diff
@@ -0,0 +1,428 @@
+	if err != nil {
+		return errors.Wrapf(err, "Failed to parse inline configuration at %s", sourcePath)
+	}
+
+	configureBlock, found := blocks["configure"]
+	if !found {
+		g.logger.DebugWith("No configure block found in source code, returning empty config", "sourcePath", sourcePath)
+
+		return nil
+	}
+
+	unmarshalledInlineConfigYAML, found := configureBlock["function.yaml"]
+	if !found {
+		return errors.Errorf("No function.yaml file found inside configure block at %s", sourcePath)
+	}
+
+	// must use yaml.v2 here since yaml.Marshal will err (not sure why)
+	marshalledYAMLContents, err := yamlv2.Marshal(unmarshalledInlineConfigYAML)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to marshal inline config from source at %s", sourcePath)
+	}
+
+	if err = yaml.Unmarshal(marshalledYAMLContents, configuration); err != nil {
+		return errors.Wrapf(err, "Failed to unmarshal inline config from source at %s", sourcePath)
+	}
+
+	return nil
+}
+
+func (g *Generator) writeOutputFile(functionTemplates []*functiontemplates.FunctionTemplate) error {
+	g.logger.DebugWith("Writing output file", "path", g.outputPath, "numFunctions", len(functionTemplates))
+
+	outputFile, err := os.Create(g.outputPath)
+	if err != nil {
+		return errors.Wrap(err, "Failed to create output file")
+	}
+
+	defer outputFile.Close()
+
+	err = packageTemplate.Execute(outputFile, struct {
+		FunctionTemplates  []*functiontemplates.FunctionTemplate
+		FunctionsByRuntime map[string][]string
+	}{
+		FunctionTemplates:  functionTemplates,
+		FunctionsByRuntime: g.functions,
+	})
+
+	if err != nil {
+		return errors.Wrap(err, "Failed to execute template")
+	}
+
+	outputFileInfo, err := outputFile.Stat()
+	if err != nil {
+		return errors.Wrap(err, "Failed to stat output file")
+	}
+
+	g.logger.InfoWith("Output file written successfully", "len", outputFileInfo.Size())
+
+	return nil
+}
+
+func createLogger() (logger.Logger, error) {
+	return nucliozap.NewNuclioZapCmd("generator", nucliozap.DebugLevel)
+}
+
+func createGenerator(logger logger.Logger, examplesDir string, outputPath string) (*Generator, error) {
+	newGenerator := Generator{
+		logger:      logger,
+		examplesDir: examplesDir,
+		outputPath:  outputPath,
+	}
+
+	// TODO: support java parser too i guess
+	//newGenerator.runtimes = []string{"golang", "python", "pypy", "nodejs", "java", "dotnetcore", "shell"}
+
+	newGenerator.runtimes = []string{"golang", "python", "pypy", "nodejs", "dotnetcore", "shell"}
+
+	slashSlashParser := inlineparser.NewParser(logger, "//")
+	poundParser := inlineparser.NewParser(logger, "#")
+
+	newGenerator.inlineParsers = map[string]*inlineparser.InlineParser{
+		"golang":     slashSlashParser,
+		"python":     poundParser,
+		"pypy":       poundParser,
+		"nodejs":     slashSlashParser,
+		"dotnetcore": slashSlashParser,
+		"shell":      poundParser,
+	}
+
+	newGenerator.functions = map[string][]string{}
+
+	return &newGenerator, nil
+}
+
+func main() {
```

Comment:
Convention is `newX` to create something... so would create the logger here and then call `newGenerator` (which is `createGenerator`, then call `generate` on that instance (would kill the notion of `Run`).